publishing/datasetAPI/get_datasets_test.go

```diff
@@ -48,32 +49,13 @@ func TestSuccessfulGetAListOfDatasets(t *testing.T) {
 	datasetAPI := httpexpect.New(t, cfg.DatasetAPIURL)
 
 	Convey("Get a list of datasets", t, func() {
-		Convey("when the user is unauthorised", func() {
-			response := datasetAPI.GET("/datasets").
-				Expect().Status(http.StatusOK).JSON().Object()
-
-			response.Value("items").Array().Element(0).Object().Value("id").NotNull()
-
-			for i := 0; i < len(response.Value("items").Array().Iter()); i++ {
-				//Unauthorised user so should NOT have an unpublished dataset in response
-				response.Value("items").Array().Element(i).Object().Value("id").NotEqual("133")
-
-				if response.Value("items").Array().Element(i).Object().Value("id").String().Raw() == datasetID {
-					// check the published test dataset document has the expected returned fields and values
-					response.Value("items").Array().Element(i).Object().Value("id").Equal(datasetID)
-					checkDatasetResponse(datasetID, response.Value("items").Array().Element(i).Object())
-				}
+		Convey("when the user is authorised", func() {
 
-				if response.Value("items").Array().Element(i).Object().Value("id").String().Raw() == "133" {
-					// user is not authenticated to see this item, if it is returned force failure
-					t.Log("user is not authenticated to see this item, forced test failure")
-					t.Fail()
-				}
-			}
-		})
+			expectedDatasets := make(map[string]int)
```

Comment:
why is this a map and not just a count? the same incrementing could happen in the for loop

Reply:
So firstly yes it could just be a count of int type and secondly we wouldn't want it in the for loop as some people do not use the test database to set up test data when running tests, and hence they could have more than two datasets being returned, hence we need to look for the two datasets that we are inspecting/checking. 

Also if someone does not teardown there test data between scenarios correctly this could give a false negative here.

I will change variable to an int tho